Codebase has many pylint errors

### Background / motivation

Codebase has many pylint errors. Somehow remote CI isn't catching it.

### TODO
- [x] get CI to catch them, if possible. Update: it already has it, there's no obvious way to change it
- [ ] fix the errors

### Appendix: Errors
```text
(venv) trentmc@tlm-macbook: ~/code/df-py $ pylint *
************* Module df_py.util.http_provider
df_py/util/http_provider.py:21:47: E1134: Non-mapping value self.get_request_kwargs() is used in a mapping context (not-a-mapping)
************* Module df_py.util.dftool_module
df_py/util/dftool_module.py:1075:14: E1120: No value for argument 'self' in unbound method call (no-value-for-parameter)
************* Module df_py.util.request
df_py/util/request.py:21:17: I1101: Module 'lru' has no 'LRU' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
************* Module df_py.util.oceantestutil
df_py/util/oceantestutil.py:251:11: E1120: No value for argument 'self' in unbound method call (no-value-for-parameter)
df_py/util/oceantestutil.py:256:8: E1120: No value for argument 'self' in unbound method call (no-value-for-parameter)
************* Module df_py.util.oceanutil
df_py/util/oceanutil.py:359:14: E1120: No value for argument 'values' in unbound method call (no-value-for-parameter)
************* Module df_py
df_py.egg-info/__init__.py:1:0: F0010: error while code parsing: Unable to load file df_py.egg-info/__init__.py:
[Errno 2] No such file or directory: 'df_py.egg-info/__init__.py' (parse-error)
```

--- a/.github/workflows/pylint.yml
+++ b/.github/workflows/pylint.yml
@@ -28,4 +28,4 @@ jobs:
           
       - name: Run Pylint
         run: |
-          pylint --rcfile .pylintrc *
+          pylint --recursive=true --rcfile .pylintrc *

--- a/df_py/util/dftool_module.py
+++ b/df_py/util/dftool_module.py
@@ -1072,7 +1072,10 @@ def _getSecretSeedOrExit() -> int:
 def _getPrivateAccount():
     private_key = os.getenv("DFTOOL_KEY")
     assert private_key is not None, "Need to set envvar DFTOOL_KEY"
+
+    # pylint: disable=no-value-for-parameter
     account = Account.from_key(private_key=private_key)
+
     print(f"For private key DFTOOL_KEY, address is: {account.address}")
     return account
 

--- a/df_py/util/http_provider.py
+++ b/df_py/util/http_provider.py
@@ -17,6 +17,8 @@ class CustomHTTPProvider(HTTPProvider):
             "Making request HTTP. URI: %s, Method: %s", self.endpoint_uri, method
         )
         request_data = self.encode_rpc_request(method, params)
+
+        # pylint: disable=not-a-mapping
         raw_response = make_post_request(
             self.endpoint_uri, request_data, **self.get_request_kwargs()
         )

--- a/df_py/util/oceantestutil.py
+++ b/df_py/util/oceantestutil.py
@@ -248,11 +248,14 @@ def random_lock_and_allocate(web3, tups: list):
 
 
 def get_account0():
-    return Account.from_key(private_key=os.getenv("TEST_PRIVATE_KEY0"))
+    return _account(0)
 
 
 def get_all_accounts():
-    return [
-        Account.from_key(private_key=os.getenv(f"TEST_PRIVATE_KEY{index}"))
-        for index in range(0, 9)
-    ]
+    return [_account(index) for index in range(9)]
+
+
+# pylint: disable=no-value-for-parameter
+def _account(index: int):
+    private_key = os.getenv(f"TEST_PRIVATE_KEY{index}")
+    return Account.from_key(private_key=private_key)

--- a/df_py/util/oceanutil.py
+++ b/df_py/util/oceanutil.py
@@ -356,7 +356,7 @@ def get_zero_provider_fee_dict(web3: Web3, provider_account) -> Dict[str, Any]:
     provider_fee_token = ZERO_ADDRESS
     valid_until = 0
 
-    message = Web3.solidity_keccak(
+    message = Web3.solidity_keccak(  # pylint: disable=no-value-for-parameter
         ["bytes", "address", "address", "uint256", "uint256"],
         [
             Web3.to_hex(Web3.to_bytes(text=provider_data)),

--- a/df_py/util/request.py
+++ b/df_py/util/request.py
@@ -18,6 +18,7 @@ def _remove_session(key, session):
     session.close()
 
 
+# pylint: disable=c-extension-no-member
 _session_cache = lru.LRU(8, callback=_remove_session)
 
 

